File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/GremlinNob_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";


export class GremlinNob_Model extends MonsterInfo{
    public static Id = "GremlinNob"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(85, 90);
        else
            return Mathf.RandomRange(82, 86);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/nobGremlin/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/nobGremlin/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/nobGremlin/skeleton.png"));
    }

    get Name(): string {
        return "地精大块头";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/GremlinNob_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class GremlinNob_Model : MonsterInfo
    {
        public static string Id = "GremlinNob";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(85, 90) : Random.Range(82,86);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/nobGremlin/skeleton.atlas",
                    "Monsters/theBottom/nobGremlin/skeleton.json", "Monsters/theBottom/nobGremlin/skeleton.png");
            }
        }

        public override string Name => "地精大块头";

        public GremlinNob_Model(Vector2 pos) : base(pos)
        {
        }
    }
}